common/gossmap.c

```diff
@@ -414,11 +414,8 @@ static struct gossmap_chan *add_channel(struct gossmap *map,
 	 * that's the only time we get duplicates */
 	scid.u64 = map_be64(map, cannounce_off + plus_scid_off);
 	chan = gossmap_find_chan(map, &scid);
-	if (chan) {
-		assert(chan->private);
-		assert(!private);
```

Comment:
Do we know the cause of the duplicate private channel announcement? Maybe log this case with status_broken for visibility.